modcc/module.cpp

```diff
@@ -40,6 +40,15 @@ Module::Module(std::vector<char> const& buffer)
         buffer_.push_back(0);
 }
 
+Module::Module(const char* buffer, size_t count)
+{
```

Comment:
> On the performance side, calculating the size of the buffer to be inserted before inserting means that we have fewer malloc than with push_back one by one.

👍  That's a valid argument. I would go for it then + the reserve before pushing back. Rangifying it would be fun, but I will need to rewrite [this](https://github.com/eth-cscs/nestmc-proto/blob/master/tests/unit/test_range.cpp#L146) unless we move it to util. But again, that won't fix the possible reallocation you just mentioned. So I'll go with your original suggestion + the `reserve()`.